### PR TITLE
DellEMC S6100: Update reboot-cause determination to support new firmware

### PR DESCRIPTION
**Pull request by ArunSaravananBalachandran**
**- Why I did it**

For determining reboot-cause while running newer BIOS, SMF firmware.

**- How I did it**

Made changes in reboot-cause determination script to add support for behavior of newer firmware.

**- How to verify it**

Performed different type of resets and verified "show reboot-cause" provides the correct reason.
Logs: [UT_logs.txt](https://github.com/Azure/sonic-buildimage/files/5487072/UT_logs.txt)

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [x] 201911
- [x] 202006

**- Description for the changelog**

DellEMC S6100: Update reboot-cause determination to support new firmware

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: platform/broadcom/sonic-platform-modules-dell/s6100/scripts/track_reboot_reason.sh [hunk 1]
```diff
@@ -143,27 +144,46 @@ update_mailbox_register(){
...
-            echo 0xbb > $MAILBOX_POWERON_REASON
-        elif [[ $is_thermal_reboot = 1 ]]; then
-            echo 0xee > $MAILBOX_POWERON_REASON
-        elif [[ $is_wd_reboot = 1 ]] && [[ $reason != "cc" ]]; then
-            echo 0xdd > $MAILBOX_POWERON_REASON
-        elif [[ $reason = "cc" ]]; then
-            echo 0xaa > $MAILBOX_POWERON_REASON
+
+        SMF_FPGA_VERSION_MAJOR=$(( (SMF_FPGA_VERSION & 0xF0) >> 4 ))
+        SMF_FPGA_VERSION_MINOR=$(( SMF_FPGA_VERSION & 0x0F ))
```
Comment (sujinmkang) [inline platform/broadcom/sonic-platform-modules-dell/s6100/scripts/track_reboot_reason.sh:154]: Do you store the SMF/FPGA VERSION together? Is there any chance to have this version with only upgrading either the SMF or the FPGA firmware?

Comment (ArunSaravananBalachandran) [reply platform/broadcom/sonic-platform-modules-dell/s6100/scripts/track_reboot_reason.sh:158]: During firmware upgrade, SMF FPGA will be upgraded only after SMF MSS has been updated to the newer version. Hence the earlier check was to verify that the running SMF FPGA was the newer one.

A new commit has been added to verify that both running SMF MSS, FPGA are newer ones for completeness.

**Comment by ArunSaravananBalachandran** [comment]
retest mellanox please

**Comment by ArunSaravananBalachandran** [comment]
retest vs please

**Review by sujinmkang — APPROVED** [review]

**Pull request merged**